ste/xfer-anyToRemote-file.go

```diff
@@ -171,10 +171,18 @@ func anyToRemote(jptm IJobPartTransferMgr, p pipeline.Pipeline, pacer pacer, sen
 
 		if srcURL.Hostname() == dstURL.Hostname() &&
 			srcURL.EscapedPath() == dstURL.EscapedPath() {
-			jptm.LogSendError(info.Source, info.Destination, "Transfer source and destination are the same, which would cause data loss. Aborting transfer.", 0)
-			jptm.SetStatus(common.ETransferStatus.Failed())
-			jptm.ReportTransferDone()
-			return
+
+			srcRQ := srcURL.Query()
+			dstRQ := dstURL.Query()
+			if (len(srcRQ["versionId"]) > 0 || len(srcRQ["versionid"]) > 0) && !(len(dstRQ["versionId"]) > 0 || len(dstRQ["versionid"]) > 0) {
+				// Case: Replacing the current version of the blob with the previous version.
+				// In this particular case, source URL should contain version id and destination URL should not have any version id specified
```

Comment:
suggestion: if destRQ has a version ID, we should throw a meaningful error to tell the user that they cannot copy into a version explicitly.

Reply:
Ohh Okay. I'll add that.